plugins/inputs/lanz_consumer/lanz_consumer.go

```diff
@@ -0,0 +1,154 @@
+package lanz_consumer
+
+import (
+	"net/url"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/aristanetworks/goarista/lanz"
+	pb "github.com/aristanetworks/goarista/lanz/proto"
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+)
+
+var sampleConfig = `
+  	## URL to Arista LANZ endpoint
+  	# servers = "tcp://localhost:50001"
+  	servers = [
+		"tcp://switch02.int.example.com:50001",
+		"tcp://switch02.int.example.com:50001"
+	]
+`
+
+func init() {
+	inputs.Add("lanz_consumer", func() telegraf.Input {
+		return NewLanzConsumer()
+	})
+}
+
+type LanzConsumer struct {
+	Servers []string
+	Clients map[string]*LanzClient
+}
+
+func NewLanzConsumer() *LanzConsumer {
+	return &LanzConsumer{}
+}
+
+func (l *LanzConsumer) SampleConfig() string {
+	return sampleConfig
+}
+
+func (l *LanzConsumer) Description() string {
+	return "Read metrics off Arista LANZ, via socket"
+}
+
+func (l *LanzConsumer) Gather(acc telegraf.Accumulator) error {
+	return nil
+}
+
+func (l *LanzConsumer) Start(acc telegraf.Accumulator) error {
+
+	if len(l.Servers) == 0 {
+		l.Servers = append(l.Servers, "tcp://127.0.0.1:50001")
+	}
+
+	for _, server := range l.Servers {
+		c := NewLanzClient()
+		c.Server = server
+		c.Start(acc)
+	}
+	return nil
+}
+
+func (l *LanzConsumer) Stop() {
+	for _, client := range l.Clients {
+		client.Stop()
+	}
+}
+
+type LanzClient struct {
+	Server string
+	in     chan *pb.LanzRecord
+	done   chan bool
+	acc    telegraf.Accumulator
+	client *lanz.Client
+	sync.Mutex
+}
+
+func NewLanzClient() *LanzClient {
+	return &LanzClient{}
+}
+
+func (c *LanzClient) Start(acc telegraf.Accumulator) error {
+	c.acc = acc
+	deviceUrl, err := url.Parse(c.Server)
+	if err != nil {
+		return err
+	}
+	client := lanz.New(lanz.WithAddr(deviceUrl.Host), lanz.WithBackoff(1*time.Second), lanz.WithTimeout(10*time.Second))
+	c.client = &client
+	c.in = make(chan *pb.LanzRecord)
+	c.done = make(chan bool)
+	go func() {
+		client.Run(c.in)
+		c.done <- true
+	}()
+	go c.receiver(deviceUrl)
+	return nil
+}
+
+func (c *LanzClient) receiver(deviceUrl *url.URL) {
+
+	for {
+
+		select {
+		case <-c.done:
+			return
+		case msg := <-c.in:
+			cr := msg.GetCongestionRecord()
+			if cr != nil {
+				vals := map[string]interface{}{
+					"timestamp":        int64(cr.GetTimestamp()),
```

Comment:
It isn't the receive time. it is the time reported internally by the switch. we use it when diagnosing issues occasionally to compare it with logs generated to syslog

Reply:
Yes, right now we have a integer field with the switches timestamp and the metric timestamp is the time we receive it in Telegraf.  It is often useful to have these both of these timestamps, one of the advantages is that you can easily calculate the differences between the switch timestamps.

However, for selecting by time range or grouping by time for aggregation it is better supported for the metric timestamp, so another way it could make sense to handle this is to use the switches timestamp as the metric timestamp.

Both of these layouts are valid and have their own advantages/disadvantages, so I just wanted to point out that there is another option to consider.